### ISSUE TITLE
Error opening Excel file

### ISSUE THREAD

**Issue by EsbenRud** [EPPlusSoftware/EPPlus#354]
I receive error when I open the attached excel file containing Pivot-tables. I have attached the error-popup too.
The error occurs with both EPPlus 5.5.0 and 5.6.1.
Please advice in this matter.
[CapacityServerStatus-20210406.xlsx](https://github.com/EPPlusSoftware/EPPlus/files/6262186/CapacityServerStatus-20210406.xlsx)
![image](https://user-images.githubusercontent.com/12291383/113661388-4584ca00-96a6-11eb-8416-57273f2136de.png)

**Comment by EsbenRud** [comment]
Sorry, I forgot to mention that the Excel file used to work without problems.
Best regards
Esben Rud Nielsen, Bankdata

**Comment by JanKallman** [comment]
I can see the problem in the pivot table cache, but I can't reproduce it. Can you provide some code how to replicate it?

**Comment by EsbenRud** [comment]
The code is part of a WebApi Project that ain't possible to ship to you.
If I create a Excel file with the four data-source sheets and save it, then create a sample program ConsoleApp1 that makes the Overblik sheet then everything work just fine as it should.
In the real life the four data-source sheet is loaded in the same unit-of-work as the Overblik sheet, this must be a difference to the way the sample program do it.
I have attached the ConsoleApp1.
The code in trouble is in line 35 - 46 in method MakeOverviewSheet in Program.cs.
[ConsoleApp1.zip](https://github.com/EPPlusSoftware/EPPlus/files/6265882/ConsoleApp1.zip)

**Comment by JanKallman** [comment]
This sample works as you say fine. The problem occures as the shared items has a wrong data type in the incorrect file:
Correct has `s` elements:
```xml
<cacheField numFmtId="0" name="DiskType">
<sharedItems>
<s v="Silver"/>
<s v="Gold"/>
<s v="Bronze"/>
</sharedItems>
```
Incorrect has `n` elements:
```xml
<cacheField name="DiskType" numFmtId="0">
<sharedItems containsInteger="1" containsNumber="1" containsString="0" containsSemiMixedTypes="0">
<n v="Silver"/>
<n v="Gold"/>
<n v="Bronze"/>
</sharedItems>
</cacheField>
```
Do you use a template that overrides the fields on the pivot table in any way (with an integer field)?
I'll continue to investigate, but it will be hard to find the issue without a reproducable sample.

**Comment by EsbenRud** [comment]
I simply use a template with an empty "Overblik" sheet as the only sheet.
The 4 data-sheets are added and the Overblik is updated with the pivot as you can see in line 35 - 46.
I will try to create a failing-sample-project and let you hear when I have succeeded in doing that.

**Comment by EsbenRud** [comment]
Hello again, I think that I have found the reason for the failure.
The field DiskType comes from an Enum field with Gold = 1, Silver = 2 and Bronze = 3.
When I write the value to the cell then I use sheet.SetValue(row, column, enum);
If I do a 'value = value.ToString();' just before the call to SetValue then everything works fine.
Does this give any meaning for you?

**Comment by EsbenRud** [comment]
Sorry, correction, if I do a 'enum = enum.ToString();' just before ....

**Comment by JanKallman** [comment]
Yes, it does. Interesting. Here is the code that writes the shared items.
https://github.com/EPPlusSoftware/EPPlus/blob/82c5ec35501ce8c7974fc0072849f45e26756fff/src/EPPlus/Table/PivotTable/ExcelPivotTableCacheField.cs#L296
Most likley the enum is identified as an integer but converted into a string causing the error. I hope I can add a fix shortly, if I can reproduce the issue. Thanks for the feedback

**Comment by JanKallman** [comment]
This should be fixed in 5.6.2. I'll close this issue. If you have further problem, please reopen this issue
